lib/router/src/lib.rs

```diff
@@ -117,14 +115,14 @@ where
     Mk::Value: svc::Service<Req> + Clone,
 {
     pub fn new(recognize: Rec, make: Mk, capacity: usize, max_idle_age: Duration) -> Self {
-        let (cache, _cache_bg) = Cache::new(capacity, max_idle_age);
+        let (cache, cache_bg) = Cache::new(capacity, max_idle_age);
 
         Router {
             inner: Arc::new(Inner {
                 recognize,
                 make,
                 cache,
-                _cache_bg,
+                cache_bg,
```

Comment:
I don't really get why the cache purging task is a field on `Inner` --- I think it makes more sense to have `Router::new` return `(Router, PurgeCache)` instead, so that the caller can spawn the purging task...